**Inserting Libraries**

In [ ]:
import numpy as np
import pandas as pd
import itertools
from sklearn.model_selection import train_test_split
from sklearn.feature_extraction.text import TfidfVectorizer
from sklearn.linear_model import PassiveAggressiveClassifier
from sklearn.metrics import accuracy_score, confusion_matrix

**Reading and Loading the Data**

In [ ]:
#Read the data
df = pd.read_csv('news.csv')
df.shape
df.head(5)

Unnamed: 0  ... label
0        8476  ...  FAKE
1       10294  ...  FAKE
2        3608  ...  REAL
3       10142  ...  FAKE
4         875  ...  REAL

[5 rows x 4 columns]

In [ ]:
#DataFlair - Get the labels
labels = df.label
labels.head()

0    FAKE
1    FAKE
2    REAL
3    FAKE
4    REAL
Name: label, dtype: object

**Splitting the Dataset**

In [ ]:
#DataFlair - Split the dataset
x_train, x_test, y_train, y_test = train_test_split(df['text'], labels, test_size=0.2, random_state=7)

1. Intilizing With English Stop Words
2. Applying Fit and Transform

In [ ]:
#DataFlair - Initialize a TfidVectorizer
tfidf_vectorizer = TfidfVectorizer(stop_words='english', max_df= 0.7)

#DataFlair - Fit and Transform train set, transform train set

tfidf_train = tfidf_vectorizer.fit_transform(x_train)
tfidf_test = tfidf_vectorizer.transform(x_test)

**Initializing a PassiveAggressiveClassifier and Predicting on Test Set**

In [ ]:
#DataFlair - Initialize a PassiveAggressiveClassifier
pac = PassiveAggressiveClassifier(max_iter= 50)
pac.fit(tfidf_train, y_train)

#DataFlair - Predict on the test set and calculate accuracy
y_pred = pac.predict(tfidf_test)
score = accuracy_score(y_test, y_pred)
print(f'Accuracy: {round(score*100,2)}%')

Accuracy: 92.82%


**Confusion Matrix**

In [ ]:
#DataFlair - Build Confusion Matrix
confusion_matrix(y_test, y_pred, labels = ['FAKE','REAL'])

array([[589,  49],
       [ 42, 587]])


*   589 = True Positive 
*   587 = True Negative


*   42 = False Positive
*   49 = False Negative



